src/d3a/models/strategy/external_strategies/load.py

```diff
@@ -400,8 +422,10 @@ def event_tick(self):
             if req.request_type == "set_energy_forecast":
                 self._set_energy_forecast_impl(req.arguments, req.response_channel)
 
-        self.pending_requests = [req for req in self.pending_requests
-                                 if req.request_type not in "set_energy_forecast"]
+        self.pending_requests = deque(
```

Comment:
sorry if its already discussed somewhere, can you explain the specific reason of having a deque here?

Reply:
Suppose I (as a user) send 4 bids (1 is the oldest and 4 is the most recent):
- Bid 1 (replace_existing: True)
- Bid 2 (replace_existing: True)
- Bid 3 (replace_existing: False)
- Bid 4 (replace_existing: True)

I expect that the most recent bid I sent (4) will replace all the others because it has `replace_existing=True` and it's the last order I sent. However, using a normal list we would have a LIFO behavior and what will happen is that:
- (4) will be the first bid request to be parsed, so it won’t find any existing bids in the market (and it won’t replace anything).
- Then (3) will be parsed, and it won’t replace anything because it has replace_existing=False (so now we have bids (4) and (3) in the market).
- Then (2) will be parsed. Since it has replace_existing=True, it will delete the bids (4) and (3) from the market. However, those bids are the most recent ones, so I would be expecting the opposite behavior.

That's why using a FIFO object like deque helps to keep the right order.

Reply:
Thanks a verbose explanation, totally makes sense now